clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -1190,6 +1195,15 @@ void handleResponse(AbstractResponse abstractResponse) {
             }
 
             @Override
+            boolean handleUnsupportedVersionException(UnsupportedVersionException exception) {
+                if (supportsDisablingTopicCreation) {
+                    supportsDisablingTopicCreation = false;
```

Comment:
This works for now since we only support disabling auto topic creation in the latest version. However, if we bump up the version of metadata request again in the future, this check may be too restrictive. If this requires more changes, perhaps we can at least add a comment for now.